File: topics/02-hashing/problems/two-sum/solution.ts
export function twoSum(nums: number[], target: number): number[] {
  const seen = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const need = target - nums[i];
    const seenIndex = seen.get(need);
    if (seenIndex !== undefined) {
      return [seenIndex, i];
    }
    seen.set(nums[i], i);
  }
  return [];
}

File: topics/02-hashing/problems/two-sum/solution.py
from typing import List


def two_sum(nums: List[int], target: int) -> List[int]:
    seen: dict[int, int] = {}
    for i, value in enumerate(nums):
        need = target - value
        if need in seen:
            return [seen[need], i]
        seen[value] = i
    return []
